### DISCUSSION TITLE
How to get the original data passed to the table

### DISCUSSION THREAD

**Discussion by shaiRos** [KevinVandy/material-react-table#574] [Q&A] [answered]
I am trying to create an action button on the toolbarcustom actions and I am passing the table instance from the param i get from this option when creating the table
`
<MaterialReactTable
...
renderTopToolbarCustomActions={({ table }) => { ... }
/>
`

I want to be able to click the action button. and grab the id's of ALL data in the table. I have tried
`table.getRowModel().rows`
but this only grabs the rows that are visible in the table and not the other rows that are in other pages. Been going through the documentation, but I can't find any methods for it. I know I can just grab the original data that I passed to MaterialReactTable but I want to be able to use the table to grab it instead because I define the id attribute names differently. Would appreciate the help if anyone knows.

**Comment by KevinVandy** [comment, marked as answer]
There are other row models like prePagination and preFiltered

**Reply by shaiRos** [reply]
thank you. prepagination model was the one I needed.
